Silicon/AlderlakePkg/Library/BaseSerialIoI2cLib/SerialIoI2cPrivateLib.c

```diff
@@ -0,0 +1,939 @@
+
+  @retval     TRUE        Tx FIFO is empty
+  @retval     FALSE       Tx FIFO not empty
+**/
+BOOLEAN
+SerialIoI2cIsTxFifoEmpty (
+  UINTN   MmioBase
+  )
+{
+  SERIAL_IO_I2C_MEM_STATUS     MemStatus;
+  MemStatus.Data32 = MmioRead32 (MmioBase + R_I2C_MEM_STATUS);
+  return (BOOLEAN) MemStatus.Fields.TFE;
+}
+
+/**
+  I2C Activity Status Check
+
+  @param[in]  MmioBase    MMIO Base Address of specified I2C controller
+
+  @retval     TRUE        I2C is Active
+  @retval     FALSE       I2C is not Active
+**/
+BOOLEAN
+STATIC
+SerialIoI2cIsIcStatusAct (
+  UINTN   MmioBase
+  )
+{
+  SERIAL_IO_I2C_MEM_STATUS     MemStatus;
+  MemStatus.Data32 = MmioRead32 (MmioBase + R_I2C_MEM_STATUS);
+  return (BOOLEAN) MemStatus.Fields.IcStatusAct;
+}
+
+/**
+  Dumps all Tx Abort Source register fields
+
+  @param[in]  MmioBase    MMIO Base Address of specified I2C controller
+**/
+VOID
+SerialIoI2cDumpTxAbort (
+  UINTN   MmioBase
+  )
+{
+  SERIAL_IO_I2C_MEM_TX_ABRT_SOURCE  TxAbortSource;
+  TxAbortSource.Data32 = MmioRead32 (MmioBase + R_I2C_MEM_TX_ABRT_SOURCE);
+
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.Abrt7bAddrNoAck %d\n", TxAbortSource.Fields.Abrt7bAddrNoAck));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.Abrt10bAddr1NoAck %d\n", TxAbortSource.Fields.Abrt10bAddr1NoAck));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.Abrt10bAddr2NoAck %d\n", TxAbortSource.Fields.Abrt10bAddr2NoAck));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtTxDataNoAck %d\n", TxAbortSource.Fields.AbrtTxDataNoAck));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtGCallNoAck %d\n", TxAbortSource.Fields.AbrtGCallNoAck));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtGCallRead %d\n", TxAbortSource.Fields.AbrtGCallRead));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtHsAckDet %d\n", TxAbortSource.Fields.AbrtHsAckDet));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtSbAckDet %d\n", TxAbortSource.Fields.AbrtSbAckDet));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtHsNoRst %d\n", TxAbortSource.Fields.AbrtHsNoRst));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtSbNoRst %d\n", TxAbortSource.Fields.AbrtSbNoRst));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.Abrt10bRdNoRst %d\n", TxAbortSource.Fields.Abrt10bRdNoRst));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtMasterDis %d\n", TxAbortSource.Fields.AbrtMasterDis));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.ArbLost %d\n", TxAbortSource.Fields.ArbLost));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtSlFlushTxFifo %d\n", TxAbortSource.Fields.AbrtSlFlushTxFifo));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtSlvArbLost %d\n", TxAbortSource.Fields.AbrtSlvArbLost));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtSlvRdIntx %d\n", TxAbortSource.Fields.AbrtSlvRdIntx));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.AbrtUserAbrt %d\n", TxAbortSource.Fields.AbrtUserAbrt));
+  DEBUG ((DEBUG_ERROR, "TxAbortSource.Fields.TxFlushCnt %d\n", TxAbortSource.Fields.TxFlushCnt));
```

Comment:
For these debug purpose debug log, better added between 
  DEBUG_CODE_BEGIN ();
  DEBUG_CODE_END ();